src/TimeSteppers/time_stepping_kernels.jl

```diff
@@ -6,39 +6,28 @@
 function calculate_interior_tendency_contributions!(G, arch, grid, coriolis, buoyancy, surface_waves, closure, 
                                                     U, C, pHY′, K, F, clock)
 
-    # Manually choose thread-block layout here as it's ~20% faster.
-    # See: https://github.com/climate-machine/Oceananigans.jl/pull/308
-    Nx, Ny, Nz = grid.Nx, grid.Ny, grid.Nz
-
-    if Nx == 1
-        Tx, Ty = 1, min(256, Ny)
-        Bx, By, Bz = Tx, floor(Int, Ny/Ty), Nz
-    elseif Ny == 1
-        Tx, Ty = min(256, Nx), 1
-        Bx, By, Bz = floor(Int, Nx/Tx), Ty, Nz
-    else
-        Tx, Ty = 16, 16
-        Bx, By, Bz = floor(Int, Nx/Tx), floor(Int, Ny/Ty), Nz
-    end
-
-    @launch(device(arch), threads=(Tx, Ty), blocks=(Bx, By, Bz),
-            calculate_Gu!(G.u, grid, coriolis, surface_waves, closure, U, C, K, F, pHY′, clock))
+    workgroup, worksize = work_layout(grid, :xyz)
 
-    @launch(device(arch), threads=(Tx, Ty), blocks=(Bx, By, Bz),
-            calculate_Gv!(G.v, grid, coriolis, surface_waves, closure, U, C, K, F, pHY′, clock))
+    calculate_Gu_kernel! = calculate_Gu!(device(arch), workgroup, worksize)
+    calculate_Gv_kernel! = calculate_Gv!(device(arch), workgroup, worksize)
+    calculate_Gw_kernel! = calculate_Gw!(device(arch), workgroup, worksize)
+    calculate_Gc_kernel! = calculate_Gc!(device(arch), workgroup, worksize)
 
-    @launch(device(arch), threads=(Tx, Ty), blocks=(Bx, By, Bz),
-            calculate_Gw!(G.w, grid, coriolis, surface_waves, closure, U, C, K, F, clock))
+    calculate_Gu_kernel!(G.u, grid, coriolis, surface_waves, closure, U, C, K, F, pHY′, clock)
```

Comment:
no synchronization?

Reply:
I'll use a `MultiEvent` for these kernels as suggested.